blenderproc/python/writer/BopWriterUtility.py

```diff
@@ -425,3 +463,373 @@ def write_frames(chunks_dir: str, dataset_objects: list, depths: Optional[List[n
+
+        :param chunks_dir: Path to the output directory of the current chunk.
+        :param starting_chunk_id: The first chunk_dir the writer has written to during this run.
+        :param starting_frame_id: The first frame id the writer has written during this run.
+        :param dataset: Only save annotations for objects of the specified bop dataset. Saves all object poses if
+                        undefined.
+        :param delta: Tolerance used for estimation of the visibility masks.
+        """
+        # This import is done inside to avoid having the requirement that BlenderProc depends on the bop_toolkit
+        # pylint: disable=import-outside-toplevel
+        from bop_toolkit_lib import inout, misc, visibility, renderer
+        # pylint: enable=import-outside-toplevel
+
+        width = bpy.context.scene.render.resolution_x
+        height = bpy.context.scene.render.resolution_y
+        # the renderer instance is a singleton - if we call this function twice, we obtain the same instance again here
+        ren = renderer.create_renderer(width=width, height=height, renderer_type='vispy', mode='depth')
+        # hence, we need to call the `__init__` again to properly set up the rendering
+        # pylint: disable=unnecessary-dunder-call
+        ren.__init__(width=width, height=height)
+        # pylint: enable=unnecessary-dunder-call
+
+        # add objects
+        # for numpy>=1.20, np.float is deprecated: https://numpy.org/doc/stable/release/1.20.0-notes.html#deprecations
+        np.float = float
+        dataset_objects = []
+        if dataset != "":
+            for obj in get_all_mesh_objects():
+                if "bop_dataset_name" in obj.blender_obj:
+                    # opposite to above, we write annotations also for hidden objects
+                    if obj.blender_obj["bop_dataset_name"] == dataset:
+                        dataset_objects.append(obj)
+        else:
+            # if no dataset is defined, write annotations for all objects
+            dataset_objects = get_all_mesh_objects()
+
+        for obj in dataset_objects:
+            ren.add_object(obj_id=obj.get_cp('category_id'), model_path=obj.get_cp('model_path'))
+
+        # Load scene info and ground-truth poses.
+        chunk_dirs = sorted(glob.glob(os.path.join(chunks_dir, '*')))
+        chunk_dirs = [d for d in chunk_dirs if os.path.isdir(d)]
+        chunk_dir_ids = [d.split('/')[-1] for d in chunk_dirs]
+        chunk_dirs = chunk_dirs[chunk_dir_ids.index(f"{starting_chunk_id:06d}"):]
+
+        for dir_counter, chunk_dir in enumerate(chunk_dirs):
+            last_chunk_gt_fpath = os.path.join(chunk_dir, 'scene_gt.json')
+            last_chunk_camera_fpath = os.path.join(chunk_dir, 'scene_camera.json')
+            scene_gt = _BopWriterUtility.load_json(last_chunk_gt_fpath, keys_to_int=True)
+            scene_camera = _BopWriterUtility.load_json(last_chunk_camera_fpath, keys_to_int=True)
```

Comment:
Since this is shared among `calc_gt_info()` and `calc_gt_mask()` and partly `calc_gt_coco()` can we move it in front? It should also be more efficient.

Reply:
Yes, I moved them outside! The only thing that needs to stay inside is the renderer init since the image dims are different.